### ISSUE TITLE
at least one example test for `ref struct`

### ISSUE THREAD

**Issue by dzmitry-lahoda** [SteveDunn/Vogen#54]

**Comment by SteveDunn** [comment]
Good idea! Vogen doesn't current support these as everything implements `IEquatable<T>`.

But it should support them.  We'll need to add constraints that ref structs can't have any 'conversions' (`TypeConverter`, Dapper, System.Text.Json, Newtonsoft.Json etc)
